python/cudf/cudf/utils/ioutils.py

```diff
@@ -1667,35 +1724,25 @@ def get_reader_filepath_or_buffer(
                     FutureWarning,
                 )
 
-        else:
-            if len(paths) == 0:
-                raise FileNotFoundError(
-                    f"{path_or_data} could not be resolved to any files"
-                )
-            path_or_data = [
-                BytesIO(
-                    _fsspec_data_transfer(
-                        fpath,
-                        fs=fs,
-                        mode=mode,
-                        bytes_per_thread=bytes_per_thread,
+    else:
+        # Sources are already buffers or file-like objects
+        for source in input_sources:
+            if not isinstance(source, iotypes) and is_file_like(source):
+                if isinstance(source, TextIOWrapper):
+                    source = source.buffer
+                filepaths_or_buffers.append(
+                    BytesIO(
+                        _fsspec_data_transfer(
+                            source,
+                            mode=mode,
+                            bytes_per_thread=bytes_per_thread,
+                        )
```

Comment:
So this is the thing we eventually want to run in parallel, but for now we are just refactoring...

Reply:
Yes exactly. If this refactoring seems complicated, then we can add the multi-file processing in a different way. However, this PR seems to simplify the existing logic a bit in IMO (but I'm obviously biased).